dht.go

```diff
@@ -651,14 +638,20 @@ func (dht *DHT) getLinks(base Hash, tag string, statusMask int) (results []Tagge
 			x := strings.Split(key, ":")
 			t := string(x[3])
 			if string(x[1]) == b && (tag == "" || tag == t) {
-				entry := LinkEntry{}
-				json.Unmarshal([]byte(value), &entry)
-				if err == nil && (entry.Status&statusMask) > 0 {
-					th := TaggedHash{H: string(x[2]), Source: entry.Source}
-					if tag == "" {
-						th.T = t
+				var records []LinkEvent
+				json.Unmarshal([]byte(value), &records)
+				l := len(records)
+				//TODO: this is totally bogus currently simply
+				// looking at the last item we ever got
```

Comment:
ah, this is why you can ignore the case up there when there is already a link entry, right?

Reply:
as I see it, right now it can always be just one entry in the list because you are overwriting it.

so what is the sense in this change then at all?